src/bin/pg_autoctl/pgsql.c

```diff
@@ -1182,23 +1182,27 @@ validate_connection_string(const char *connectionString)
 /*
  * pgsql_get_sync_state_and_wal_lag queries a primary PostgreSQL server to get
  * both the current pg_stat_replication.sync_state value and replication lag.
+ *
+ * currentLSN is text representation of a 64 bit LSN value. NAMEDATALEN is
+ * large enough contain the textual representation.
  */
 typedef struct PgsrSyncAndWALContext
 {
 	bool		parsedOk;
 	char		syncState[PGSR_SYNC_STATE_MAXLENGTH];
-	uint64_t	walLag;
+	char        currentLSN[NAMEDATALEN];
 } PgsrSyncAndWALContext;
 
 bool
 pgsql_get_sync_state_and_wal_lag(PGSQL *pgsql, const char *slotName,
-								 char *pgsrSyncState, int64_t *walLag,
+								 char *pgsrSyncState, char *currentLSN,
+								 int bufferLength,
```

Comment:
Let's have a better name than `buferLength` which is not only very generic but also confusing: it's not the length of the string as per `strlen` but the size of the memory allocated. What about `currentLSNsize` instead?

Reply:
ok, made id `maxLSNSize`

Reply:
and ditched -1